_locals.tf

```diff
@@ -1,4 +1,14 @@
 locals {
+  labels = merge(
+    var.labels,
+    {
+      "environment" = var.environment
+      "team"        = var.team
+    },
+  )
+
+  project = var.project_override != null ? var.project_override : var.team == "analytics" ? "treehouse-analyticsplatform" : var.environment == "shared" ? "treehouse-devops" : "treehouse-services-${var.environment}"
```

Comment:
Bit too complex and fragile to my liking. But, haven't come up with suitable alternatives, yet.

Reply:
```hcl
project = var.project_override != null ? var.project_override : lookup(local.custom_team_projects, var.team, "treehouse-services-${var.environment}")
```
and
```hcl
locals {
  custom_team_projects = {
    "analytics": "team-analyticplatform",
    "shared": "treehouse-devops",
    "global": "treehouse-devops",
  }
}
```
Maybe this makes the conditional easier to read?

Reply:
and also I don't think shared needs to goto devops, it should just go to the environment it belongs too (services production or staging)
A resource can be staging specific shared resource

Reply:
`shared` **environment** goes to `treehouse-devops`.

Which would make it something like this:
```terraform
  environment_project_overrides = {
    "global":     "treehouse-services-devops",
    "shared":     "treehouse-services-devops",
  }
  team_project_overrides = {
    "analytics": "treehouse-analyticsplatform",
  }
  project = coalesce(
    var.project_override,
    lookup(
      local.team_project_overrides,
      var.team,
      null,
    ),
    lookup(
      local.environment_project_overrides,
      var.environment,
      null,
    ),
    "treehouse-services-${var.environment}",
  )
```

Reply:
Or, even more compact:
```terraform
  environment_project_overrides = {
    "global":     "treehouse-services-devops",
    "shared":     "treehouse-services-devops",
  }
  team_project_overrides = {
    "analytics": "treehouse-analyticsplatform",
  }
  project = coalesce(
    var.project_override,
    try(
      local.team_project_overrides[var.team],
      local.environment_project_overrides[var.environment],
      "treehouse-services-${var.environment}",
    ),
  )
```